enhancements/compliance-operator.md

```diff
@@ -0,0 +1,246 @@
+* The remediations were typically done using a shell script or Ansible. For
+  OpenShift we’d want the remediations done using MachineConfigs or other
+  Custom Resources. However, the remediations are not a hard prerequisite for
+  the first version of the operator, even identifying the gaps in compliance
+  would be useful.
+* For the gaps in compliance we identify, a check must be written. These
+  checks are being compiled as a parallel effort. Having a full set of
+  checks and/or remediations is not a hard prerequisite for the operator
+  itself, though as the content would be delivered separately.
+
+These have all been filed with the OpenSCAP team and acknowledged by them.
+
+### Goals
+
+A cluster administrator is able to use the operator to assess the degree
+of compliance against a given security standard. In particular this would
+mean:
+
+* Provide a way for cluster administrator to start cluster scans
+* Inspect the scan result on the high level (pass/fail/error)
+* Provide a way to collect the detailed results of the scan
+  * The results would list a way to remediate the gaps, typically by the means
+    of deploying certain manifests to OpenShift (e.g. a Machineconfig), a different
+    way of changing cluster configuration or even a free-form recommendation text.
+
+### Non-Goals
+
+The operator itself does not perform any scans or remediations, only
+wraps the OpenSCAP scanner and provides an OpenShift-native API to
+run the scans.
+
+## Proposal
+
+An instance of a scan would be represented by a CR. The CR must include
+the scan profile (e.g. FedRAMP moderate) and the content to scan against
+(typically `ocp4`). The CR might optionally contain a set of individual rules
+(e.g. is `auditd` enabled on the nodes?) and an image with the content.
+The CR would be processed by the operator, which would launch a pod on
+each node to scan that node and deliver scan results.
+
+The pod would consist of two containers and an init container. The init
+container is started with the image that contains the content for the
+scanner and copies the content to a volume to be used by the scanner
+container. The two "worker" containers are the scanner and a log collector.
+This is because OpenSCAP writes the report to a file, so the scanner and
+the collector share a directory where the scanner writes the report, the
+collector would wait for the report to be created and write the contents
+of that file to a location where the cluster administrator can grab the
+report and inspect it.
+
+The scanner container in the pod would mount the node's file system into
+a directory and point the `openscap-chroot` binary there.
+
+### User Stories
+
+#### As an OpenShift admin, I want to assess the degree of compliance of my cluster with FedRAMP moderate.
+
+This is a use-case where the admin just wants to visualize the gaps in
+compliance. They would perform the remediation themselves, potentially
+based on the MCOs the compliance operator suggests.
```

Comment:
s/MCOs/MachineConfigs/

MCO stands for Machine-Config-Operator, so that's not what we're trying to portray here.